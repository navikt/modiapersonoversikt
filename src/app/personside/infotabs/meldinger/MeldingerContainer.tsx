import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import styled from 'styled-components';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import TraadVisningContainer from './traadvisning/TraadVisningContainer';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import VerktoylinjeContainer from './traadvisning/verktoylinje/VerktoylinjeContainer';
import TraadListe from './traadliste/TraadListe';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';

const meldingerMediaTreshold = pxToRem(900);

const MeldingerArticleStyle = styled.article`
    @media (min-width: ${meldingerMediaTreshold}) {
        display: flex;
        align-items: flex-start;
        > *:last-child {
            margin-left: ${theme.margin.layout};
        }
    }
    > * {
        margin-bottom: ${theme.margin.layout};
    }
    position: relative;
    > *:first-child {
        flex: 30% 1 1;
    }
    > *:last-child {
        position: sticky;
        top: 0;
        flex: 70% 1 1;
    }
`;

function MeldingerContainer() {
    return (
        <article>
            <RestResourceConsumer<Traad[]>
                getResource={restResources => restResources.tr√•derOgMeldinger}
                returnOnPending={<CenteredLazySpinner />}
            >
                {data => (
                    <MeldingerArticleStyle>
                        <TraadListe traader={data} />
                        <div>
                            <VerktoylinjeContainer />
                            <TraadVisningContainer />
                        </div>
                    </MeldingerArticleStyle>
                )}
            </RestResourceConsumer>
        </article>
    );
}

export default MeldingerContainer;
