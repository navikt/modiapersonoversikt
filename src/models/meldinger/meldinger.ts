import { Temagruppe } from '../Temagrupper';
import { JournalforingsSak } from '../../app/personside/infotabs/meldinger/traadvisning/verktoylinje/journalforing/JournalforingPanel';

export interface Traad {
    traadId: string;
    meldinger: Melding[];
}

export interface Melding {
    id: string;
    oppgaveId?: string;
    meldingstype: Meldingstype;
    temagruppe: Temagruppe | null;
    skrevetAvTekst: string;
    journalfortAv?: Saksbehandler;
    journalfortDato?: string;
    journalfortTema?: string;
    journalfortTemanavn?: string;
    journalfortSaksid?: string;
    fritekst: string;
    lestDato?: string;
    status: LestStatus;
    statusTekst?: string;
    opprettetDato: string;
    ferdigstiltDato?: string;
    erFerdigstiltUtenSvar: boolean;
    ferdigstiltUtenSvarAv?: Saksbehandler;
    kontorsperretEnhet?: string;
    kontorsperretAv?: Saksbehandler;
    markertSomFeilsendtAv?: Saksbehandler;
    erDokumentMelding: boolean;
    ferdigstiltUtenSvarDato?: string;
}

export interface Saksbehandler {
    fornavn: string;
    etternavn: string;
    ident?: string;
}

export enum TypeKontakt {
    Telefon = 'telefon',
    Oppmøte = 'oppmote',
    Oppgave = 'oppgave',
    Dokument = 'dokument',
    Monolog = 'monolog',
    MonologUbesvart = 'monlog ubesvart',
    Dialog = 'dialog',
    DialogBesvart = 'dialog besvart'
}

export enum Meldingstype {
    DOKUMENT_VARSEL = 'DOKUMENT_VARSEL',
    OPPGAVE_VARSEL = 'OPPGAVE_VARSEL',
    SPORSMAL_SKRIFTLIG = 'SPORSMAL_SKRIFTLIG',
    SPORSMAL_SKRIFTLIG_DIREKTE = 'SPORSMAL_SKRIFTLIG_DIREKTE',
    SVAR_SKRIFTLIG = 'SVAR_SKRIFTLIG',
    SVAR_OPPMOTE = 'SVAR_OPPMOTE',
    SVAR_TELEFON = 'SVAR_TELEFON',
    DELVIS_SVAR_SKRIFTLIG = 'DELVIS_SVAR_SKRIFTLIG',
    SAMTALEREFERAT_OPPMOTE = 'SAMTALEREFERAT_OPPMOTE',
    SAMTALEREFERAT_TELEFON = 'SAMTALEREFERAT_TELEFON',
    SPORSMAL_MODIA_UTGAAENDE = 'SPORSMAL_MODIA_UTGAAENDE',
    SVAR_SBL_INNGAAENDE = 'SVAR_SBL_INNGAAENDE'
}

export enum LestStatus {
    IkkeLest = 'IKKE_LEST_AV_BRUKER',
    Lest = 'LEST_AV_BRUKER',
    IkkeBesvart = 'IKKE_BESVART'
}

export interface SendReferatRequest {
    fritekst: string;
    temagruppe: Temagruppe;
    meldingstype: Meldingstype.SAMTALEREFERAT_TELEFON | Meldingstype.SAMTALEREFERAT_OPPMOTE;
}

export interface SendSpørsmålRequest {
    fritekst: string;
    sak: JournalforingsSak;
    erOppgaveTilknyttetAnsatt: boolean;
}

export interface ForsettDialogRequest {
    traadId: string;
    behandlingsId: string;
    fritekst: string;
    sak?: JournalforingsSak;
    erOppgaveTilknyttetAnsatt: boolean;
    meldingstype: Meldingstype;
    oppgaveId?: string;
}

export interface SendDelsvarRequest {
    traadId: string;
    behandlingsId: string;
    fritekst: string;
    temagruppe: Temagruppe;
    oppgaveId: string;
}

export interface OpprettHenvendelseRequest {
    traadId: string;
}

export interface OpprettHenvendelseResponse {
    behandlingsId: string;
    oppgaveId?: string;
}

export interface SlaaSammenRequest {
    traader: SlaaSammenTraad[];
    temagruppe: Temagruppe;
}

export interface SlaaSammenTraad {
    oppgaveId: string;
    traadId: string;
}

export interface SlaaSammenResponse {
    nyTraadId: string;
    traader: Traad[];
}
