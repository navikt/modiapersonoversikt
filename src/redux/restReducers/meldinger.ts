import { getMeldinger } from '../../api/meldinger-api';
import { createActionsAndReducer } from './restResource';

const { reducer, action, reload, tilbakestill } = createActionsAndReducer('meldinger');

export function hentMeldinger(fødselsnummer: string) {
    return action(() => getMeldinger(fødselsnummer));
}

export function reloadMeldinger(fødselsnummer: string) {
    return reload(() => getMeldinger(fødselsnummer));
}

export function resetMeldingerResource() {
    return tilbakestill;
}

export default reducer;
