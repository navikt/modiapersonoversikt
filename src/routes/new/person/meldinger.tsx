import { createFileRoute } from '@tanstack/react-router';
import { keepQueryParams } from 'src/utils/keepQueryParams';
import { z } from 'zod';

const meldingerSearchSchema = z.object({
    traadId: z.string().optional()
});

export const meldingerRouteMiddleware = keepQueryParams(['traadId']);

export const Route = createFileRoute('/new/person/meldinger')({
    validateSearch: meldingerSearchSchema,
    search: {
        middlewares: [meldingerRouteMiddleware]
    }
});
