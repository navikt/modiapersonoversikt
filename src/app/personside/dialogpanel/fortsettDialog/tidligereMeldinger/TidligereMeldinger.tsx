import * as React from 'react';
import { Traad } from '../../../../../models/meldinger/meldinger';
import styled from 'styled-components/macro';
import EnkeltMelding from './EnkeltMelding';
import { theme } from '../../../../../styles/personOversiktTheme';
import ErrorBoundary from '../../../../../components/ErrorBoundary';
import Ekspanderbartpanel from 'nav-frontend-ekspanderbartpanel';
import { Undertittel } from 'nav-frontend-typografi';
import Panel from 'nav-frontend-paneler';
import IfFeatureToggleOn from '../../../../../components/featureToggle/IfFeatureToggleOn';
import IfFeatureToggleOff from '../../../../../components/featureToggle/IfFeatureToggleOff';
import { FeatureToggles } from '../../../../../components/featureToggle/toggleIDs';
import NyEnkeltMelding from './NyEnkeltMelding';

interface Props {
    traad: Traad;
}

const StyledArticle = styled.article`
    .ekspanderbartPanel__innhold {
        padding: 0rem;
    }
    > *:not(:last-child) {
        margin-bottom: 0.5rem;
    }
    margin-top: 0.7rem;
`;

const StyledLi = styled.li`
    position: relative;
`;

const StyledPanel = styled(Panel)`
    padding: 0rem;
    margin-top: 0.1rem;
    > li:not(:last-child) {
        margin-bottom: 0.1rem;
    }
    > li:not(:only-child) {
        border-top: ${theme.border.skilleSvak};
    }
`;

function Traadpanel(props: { traad: Traad; tittel: string; defaultApen: boolean }) {
    if (props.traad.meldinger.length === 0) {
        return null;
    }

    const flereMeldinger = props.traad.meldinger.length > 1;
    const meldinger = props.traad.meldinger.map((melding, index) => {
        const meldingnummer = props.traad.meldinger.length - index;

        return (
            <StyledLi key={melding.id}>
                <IfFeatureToggleOn toggleID={FeatureToggles.useNewDialogComponents}>
                    <NyEnkeltMelding
                        traadType={props.traad.traadType}
                        melding={melding}
                        erEnkeltstaende={props.traad.meldinger.length === 1}
                        defaultApen={props.defaultApen && !flereMeldinger}
                        meldingsNummer={meldingnummer}
                    />
                </IfFeatureToggleOn>
                <IfFeatureToggleOff toggleID={FeatureToggles.useNewDialogComponents}>
                    <EnkeltMelding
                        melding={melding}
                        erEnkeltstaende={props.traad.meldinger.length === 1}
                        defaultApen={props.defaultApen && !flereMeldinger}
                        meldingsNummer={meldingnummer}
                    />
                </IfFeatureToggleOff>
            </StyledLi>
        );
    });

    if (flereMeldinger) {
        return (
            <Ekspanderbartpanel
                renderContentWhenClosed={true}
                apen={false}
                tittel={<Undertittel>{props.tittel}</Undertittel>}
                border={true}
            >
                <StyledPanel>
                    <ol>{meldinger}</ol>
                </StyledPanel>
            </Ekspanderbartpanel>
        );
    } else {
        return (
            <StyledPanel aria-label={props.tittel}>
                <ol>{meldinger}</ol>
            </StyledPanel>
        );
    }
}

function TidligereMeldinger(props: Props) {
    const defaultApen = props.traad.meldinger.length === 1;

    return (
        <ErrorBoundary boundaryName="Tidligere meldinger">
            <StyledArticle>
                <Traadpanel traad={props.traad} tittel="Tidligere meldinger" defaultApen={defaultApen} />
            </StyledArticle>
        </ErrorBoundary>
    );
}

export default TidligereMeldinger;
