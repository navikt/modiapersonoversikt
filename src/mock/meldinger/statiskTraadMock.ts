import { LestStatus, Meldingstype, Traad } from '../../models/meldinger/meldinger';
import { saksbehandlerTekst } from '../../app/personside/infotabs/meldinger/utils/meldingerUtils';
import { Temagruppe } from '../../models/Temagrupper';

export const statiskTraadMock: Traad = {
    traadId: 'sg838exr',
    meldinger: [
        {
            id: 'x4k72bwd',
            meldingstype: Meldingstype.SAMTALEREFERAT_OPPMOTE,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'nzxo4q', fornavn: 'Sander', etternavn: 'Hansen' }),
            journalfortAv: { ident: 'vlzaiw', fornavn: 'Adrian', etternavn: 'Pettersen' },
            fritekst:
                'Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.',
            lestDato: '2019-07-12',
            status: LestStatus.Lest,
            opprettetDato: '2019-07-30',
            journalfortDato: '2019-07-11',
            ferdigstiltDato: '2019-07-07',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q1i6ndyj',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xmqgb8', fornavn: 'Frida', etternavn: 'Pedersen' }),
            journalfortAv: { ident: 'ly4pps', fornavn: 'Emil', etternavn: 'Nguyen' },
            fritekst:
                'Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.',
            lestDato: '2019-07-14',
            status: LestStatus.IkkeLest,
            opprettetDato: '2019-08-10',
            journalfortDato: '2019-07-21',
            ferdigstiltDato: '2019-07-16',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q57oayxa',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'e5otkk', fornavn: 'Synne', etternavn: 'Lunde' }),
            journalfortAv: { ident: '3t55yt', fornavn: 'Victoria', etternavn: 'Andresen' },
            fritekst:
                'Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.',
            lestDato: '2019-08-11',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-06',
            journalfortDato: '2019-08-12',
            ferdigstiltDato: '2019-07-12',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: '90l2nkx0',
            meldingstype: Meldingstype.DELVIS_SVAR_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xy70r8', fornavn: 'Kristoffer', etternavn: 'Fredriksen' }),
            journalfortAv: { ident: 'p0jh0d', fornavn: 'Jenny', etternavn: 'Berg' },
            fritekst:
                'Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.',
            lestDato: '2019-07-23',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-13',
            journalfortDato: '2019-07-30',
            ferdigstiltDato: '2019-07-27',
            erFerdigstiltUtenSvar: false,
            erDokumentMelding: false
        }
    ]
};
