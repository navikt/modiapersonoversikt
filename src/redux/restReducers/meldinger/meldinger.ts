import { createRestResourceReducerAndActions } from '../../../rest/utils/restResource';
import { apiBaseUri } from '../../../api/config';
import { AppState } from '../../reducers';
import { Traad } from '../../../models/meldinger/meldinger';

export function getMeldingerFetchUri(state: AppState): string {
    const fnr = state.gjeldendeBruker.f√∏dselsnummer;
    return `${apiBaseUri}/dialog/${fnr}/meldinger`;
}

export default createRestResourceReducerAndActions<Traad[]>('meldinger', getMeldingerFetchUri);
