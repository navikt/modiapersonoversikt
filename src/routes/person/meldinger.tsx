import { createFileRoute } from '@tanstack/react-router';
import MeldingerContainer from 'src/app/personside/infotabs/meldinger/MeldingerContainer';
import { keepQueryParams } from 'src/utils/keepQueryParams';
import { z } from 'zod';

const meldingerSearchSchema = z.object({
    traadId: z.string().optional()
});

export const Route = createFileRoute('/person/meldinger')({
    component: MeldingerContainer,
    validateSearch: meldingerSearchSchema,
    search: {
        middlewares: [keepQueryParams(['traadId'])]
    }
});
