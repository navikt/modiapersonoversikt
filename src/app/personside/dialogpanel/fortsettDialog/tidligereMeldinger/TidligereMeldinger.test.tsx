import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TidligereMeldinger from './TidligereMeldinger';
import { statiskTraadMock } from '../../../../../mock/meldinger/statiskTraadMock';

test('viser fortsett dialog', () => {
    const dialogPanelBody = renderer.create(<TidligereMeldinger traad={statiskTraadMock} />);

    expect(dialogPanelBody.toJSON()).toMatchSnapshot();
});
