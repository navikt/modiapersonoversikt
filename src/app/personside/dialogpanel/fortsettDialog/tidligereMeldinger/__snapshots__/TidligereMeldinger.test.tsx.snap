// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser fortsett dialog 1`] = `
.c5 p {
  margin-bottom: 0 !important;
}

.c4 {
  width: 100%;
  padding-right: 1rem;
}

.c6 {
  padding: 1.5rem 1rem 1rem;
  overflow-wrap: break-word;
  padding-top: 0;
}

.c3 .ekspanderbartPanel__innhold {
  padding: 0;
}

.c3 .ekspanderbartPanel__hode:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c3 .ekspanderbartPanel__hode:focus {
  outline: none;
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
}

.c0 {
  margin-top: 0.7rem;
}

.c0 .ekspanderbartPanel__innhold {
  padding: 0rem;
}

.c2 {
  position: relative;
  border-bottom: 1px solid #c6c2bf;
}

.c1 {
  margin-top: 0.1rem;
  border: solid 0.0625rem #78706a;
}

.c1 > li:not(:last-child) {
  margin-bottom: 0.1rem;
}

<article
  aria-labelledby="Helt tilfeldig ID"
  className="c0"
>
  <h3
    className="sr-only"
    id="Helt tilfeldig ID"
    tabIndex={-1}
  >
    Tidligere meldinger
  </h3>
  <div
    className="ekspanderbartPanel ekspanderbartPanel--lukket ekspanderbartPanel--border"
  >
    <button
      aria-expanded={false}
      className="ekspanderbartPanel__hode"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div
        className="ekspanderbartPanel__flex-wrapper"
      >
        <undertittel
          className="typo-undertittel ekspanderbartPanel__heading"
        >
          Tidligere meldinger
        </undertittel>
        <span
          className="ekspanderbartPanel__indikator"
        />
      </div>
    </button>
    <article
      aria-label="Tidligere meldinger"
      className="ekspanderbartPanel__innhold"
    >
      <ol
        aria-label="Tidligere meldinger"
        className="c1"
      >
        <li
          className="c2"
        >
          <div
            className="ekspanderbartPanel c3 ekspanderbartPanel--lukket"
          >
            <button
              aria-expanded={false}
              className="ekspanderbartPanel__hode"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <div
                className="ekspanderbartPanel__flex-wrapper"
              >
                <div
                  className="c4"
                >
                  <span
                    className="sr-only"
                  >
                    Melding 
                    3
                  </span>
                  <span
                    className="sr-only"
                  >
                    Lukket
                  </span>
                  <h4
                    className="typo-element"
                  >
                    Samtalereferat oppmøte - Arbeid
                  </h4>
                  <p
                    className="typo-undertekst"
                  >
                    30. juli 2019 00:00
                  </p>
                  <div
                    className="tekstomrade typo-normal c5"
                  >
                    <p
                      className="typo-normal blokk-xs"
                    >
                      Skrevet av Sander Hansen (nzxo4q)
                    </p>
                  </div>
                </div>
                <span
                  className="ekspanderbartPanel__indikator"
                />
              </div>
            </button>
            <article
              className="ekspanderbartPanel__innhold"
            >
              <div
                className="tekstomrade c6"
              >
                <p
                  className="typo-normal blokk-xs"
                >
                  Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.
                </p>
              </div>
            </article>
          </div>
        </li>
        <li
          className="c2"
        >
          <div
            className="ekspanderbartPanel c3 ekspanderbartPanel--lukket"
          >
            <button
              aria-expanded={false}
              className="ekspanderbartPanel__hode"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <div
                className="ekspanderbartPanel__flex-wrapper"
              >
                <div
                  className="c4"
                >
                  <span
                    className="sr-only"
                  >
                    Melding 
                    2
                  </span>
                  <span
                    className="sr-only"
                  >
                    Lukket
                  </span>
                  <h4
                    className="typo-element"
                  >
                    Spørsmål fra bruker - Arbeid
                  </h4>
                  <p
                    className="typo-undertekst"
                  >
                    10. aug. 2019 00:00
                  </p>
                </div>
                <span
                  className="ekspanderbartPanel__indikator"
                />
              </div>
            </button>
            <article
              className="ekspanderbartPanel__innhold"
            >
              <div
                className="tekstomrade c6"
              >
                <p
                  className="typo-normal blokk-xs"
                >
                  Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.
                </p>
              </div>
            </article>
          </div>
        </li>
        <li
          className="c2"
        >
          <div
            className="ekspanderbartPanel c3 ekspanderbartPanel--lukket"
          >
            <button
              aria-expanded={false}
              className="ekspanderbartPanel__hode"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <div
                className="ekspanderbartPanel__flex-wrapper"
              >
                <div
                  className="c4"
                >
                  <span
                    className="sr-only"
                  >
                    Melding 
                    1
                  </span>
                  <span
                    className="sr-only"
                  >
                    Lukket
                  </span>
                  <h4
                    className="typo-element"
                  >
                    Spørsmål fra bruker - Arbeid
                  </h4>
                  <p
                    className="typo-undertekst"
                  >
                    06. aug. 2019 00:00
                  </p>
                </div>
                <span
                  className="ekspanderbartPanel__indikator"
                />
              </div>
            </button>
            <article
              className="ekspanderbartPanel__innhold"
            >
              <div
                className="tekstomrade c6"
              >
                <p
                  className="typo-normal blokk-xs"
                >
                  Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.
                </p>
              </div>
            </article>
          </div>
        </li>
      </ol>
    </article>
  </div>
  <ol
    aria-label="Delsvar"
    className="c1"
  >
    <li
      className="c2"
    >
      <div
        className="ekspanderbartPanel c3 ekspanderbartPanel--apen"
      >
        <button
          aria-controls="Helt tilfeldig ID"
          aria-expanded={true}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <div
              className="c4"
            >
              <span
                className="sr-only"
              >
                Melding 
                1
              </span>
              <span
                className="sr-only"
              >
                Åpen
              </span>
              <h4
                className="typo-element"
              >
                Delsvar - Arbeid
              </h4>
              <p
                className="typo-undertekst"
              >
                13. aug. 2019 00:00
              </p>
              <div
                className="tekstomrade typo-normal c5"
              >
                <p
                  className="typo-normal blokk-xs"
                >
                  Skrevet av Kristoffer Fredriksen (xy70r8)
                </p>
              </div>
            </div>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
        <article
          className="ekspanderbartPanel__innhold"
        >
          <div
            className="tekstomrade c6"
          >
            <p
              className="typo-normal blokk-xs"
            >
              Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
            </p>
          </div>
        </article>
      </div>
    </li>
  </ol>
</article>
`;
